app/assets/scripts/views/analytics-index.js

```diff
@@ -10,30 +12,52 @@ var AnalyticsIndex = React.createClass({
 
   propTypes: {
     children: React.PropTypes.object,
+    _fetchProvinces: React.PropTypes.func,
     _fetchVProMMsids: React.PropTypes.func,
+    _fetchVProMMsidsProperties: React.PropTypes.func,
+    _fetchFieldVProMMsids: React.PropTypes.func,
+    _setCrossWalk: React.PropTypes.func,
+    provincesFetched: React.PropTypes.bool,
+    VProMMsFetched: React.PropTypes.bool,
+    provinces: React.PropTypes.array,
+    crosswalk: React.PropTypes.object,
+    params: React.PropTypes.object,
     VProMMSids: React.PropTypes.object,
-    params: React.PropTypes.object
+    VProMMs: React.PropTypes.object,
+    fieldIds: React.PropTypes.array,
+    fieldIdsFetched: React.PropTypes.bool
   },
 
-  render: function () {
-    let accumulator = { done: 0, total: 0 };
-    const provinceData = _.map(this.props.VProMMSids.data, (data, id) => {
-      const name = data.provinceName;
-      const done = data.vpromms.filter(v => v.inTheDatabase).length;
-      const total = data.vpromms.length;
-      accumulator.done += done;
-      accumulator.total += total;
-      const percentageComplete = (done / total).toFixed(2);
+  componentWillMount: function () {
+    this.props._fetchProvinces();
+    this.props._fetchFieldVProMMsids();
+    this.props._fetchVProMMsidsProperties();
+  },
+
+  renderAnalyticsIndex: function () {
+    const vprommsProvinceKeys = Object.keys(_.invert(this.props.crosswalk.province));
+    const provinces = this.props.provinces.filter(province => vprommsProvinceKeys.indexOf(province.id.toString()) !== -1);
+    let accumulator = { field: this.props.fieldIds.length, total: Object.keys(this.props.VProMMs).length };
+    const provinceData = _.map(provinces, (province, key) => {
+      // very very crude answer to an issue of english name, will pluck when names are cleaned;
+      const name = province.name_en.replace('Thua Thien H', 'Thua Thien');
+      const id = _.invert(this.props.crosswalk.province)[province.id];
+      const route = province.id;
+      const idTest = new RegExp(id);
```

Comment:
This is the code that you're replacing with a more proper API route slash DB query (the `LIKE` filter that I showed you), right? If so, then I'm fine with this, but as you know it's pretty inefficient and nonstandard to use regex to do simple string matching like this :)